Cilsil/Sil/ProcedureDescription.cs

```diff
@@ -112,11 +114,22 @@ public ProcedureDescription(MethodDefinition methodDefinition, Cfg cfg)
 
             Nodes = new List<CfgNode>();
             StartNode = new StartNode(location, this);
-            ExitNode = new ExitNode(Location.FromSequencePoint(methodDefinition
-                                                               .DebugInformation
-                                                               .SequencePoints
-                                                               .FirstOrDefault()),
-                                    this);
+            if (guardian)
+            {
+                ExitNode = new ExitNode(Location.FromSequencePoint(methodDefinition
```

Comment:
Instead you should assign the sequence point to a variable determined via ternary operator, for code cleanliness.